open-apparel-registry/open-apparel-registry
# Enable admins to promote a facility match to become the canonical facility

## Overview

This PR enables admins to promote a facility match to become the canonical facility. I updated the split match user interface to become `Adjust` match, and added a `Promote` button for each of the matches.

Clicking `Promote` will call the new API endpoint which updates the facility to use the list item for the selected match as its `created_from` & also updates its name, address, and location data for the new list item. It also writes an entry to the processing results log. Finally, it sends back the facility data to the adjust match screen, updating everything so that it is reflective of the new facility data.

Connects #639 

## Demo

![Screen Shot 2019-07-18 at 5 45 08 PM](https://user-images.githubusercontent.com/4165523/61494319-d623ef80-a983-11e9-8581-fa1fb258f3cf.png)

![Screen Shot 2019-07-18 at 5 45 13 PM](https://user-images.githubusercontent.com/4165523/61494320-d623ef80-a983-11e9-8560-a3251718c568.png)

## Testing Instructions

- using the standard set of facility data, sign in as an administrator
- merge some facilities together and copy the OAR ID
- paste that OAR ID into the new adjust path's component, then verify that you can
   - promote a match to become the canonical facility, seeing the changes reflected in the UI
   - continue to split facility matches using the existing apparatus.

## Checklist
